File: HHB/BJ/No_2941.java
package HHB.BJ;
import java.io.*;

public class No_2941 {
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] str = br.readLine().split("");
        int count = 0;

        for(int i=0; i<str.length; i++){
            if(str[i].equals("c") || str[i].equals("s") || str[i].equals("z") && i+1 < str.length){
                if(i+1 < str.length){
                    if(str[i+1].equals("=") || str[i+1].equals("-")) i++;
                }
            }
            else if(str[i].equals("d") && i+1 < str.length){
                if(str[i+1].equals("z")){
                    if(i+2 < str.length){
                        if(str[i+2].equals("=")) i+=2;
                    }
                }
                else if(str[i+1].equals("-")) i++;
            }
            else if(str[i].equals("l") || str[i].equals("n") && i+1 < str.length){
                if(str[i+1].equals("j")) i++;
            }

            count++;
        }
        System.out.print(count);

        br.close();
    }
}

File: HHB/BJ/No_2941.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

rl.on("line", (line) => {
    let str = line.split("");
    let count = 0;

    for(let i=0; i<str.length; i++){
        if(str[i]==="c" || str[i]==="s" || str[i]==="z" && i+1 < str.length){
            if(i+1 < str.length){
                if(str[i+1]==="=" || str[i+1]==="-") i++;
            }
        }
        else if(str[i]==="d" && i+1 < str.length){
            if(str[i+1]==="z"){
                if(i+2 < str.length){
                    if(str[i+2]==="=") i+=2;
                }
            }
            else if(str[i+1]==="-") i++;
        }
        else if(str[i]==="l" || str[i]==="n" && i+1 < str.length){
            if(str[i+1]==="j") i++;
        }
        count++;
    }
    console.log(count);

    rl.close();
}).on("close", () => {
    process.exit();
});